client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDatabaseMetaData.java

```diff
@@ -1485,6 +1519,17 @@ private static void optionalStringInFilter(List<String> filters, String columnNa
         filters.add(filter.toString());
     }
 
+    @Nullable
+    private String escapeIfNecessary(@Nullable String namePattern)
+    {
+        if (namePattern == null || !connection.isAssumeLiteralNamesInMetadataCalls()) {
+            return namePattern;
+        }
+        //noinspection ConstantConditions
+        verify(SEARCH_STRING_ESCAPE.equals("\\"));
```

Comment:
given fact we have no test for new functionality we will only notice this `verify` fails on deployed Trino, if constant value changes. A simple unit test for this method would help.